lib/widgets/message_list.dart

```diff
@@ -1481,3 +1503,77 @@ class MessageWithPossibleSender extends StatelessWidget {
         ])));
   }
 }
+
+class _EditMessageStatusRow extends StatelessWidget {
+  const _EditMessageStatusRow({
+    required this.messageId,
+    required this.status,
+  });
+
+  final int messageId;
+  final bool status;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final baseTextStyle = TextStyle(
+      fontSize: 12,
+      height: 12 / 12,
+      letterSpacing: proportionalLetterSpacing(context,
+        0.05, baseFontSize: 12));
+
+    return switch (status) {
+      // TODO parse markdown and show new content as local echo?
+      false => Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        spacing: 1.5,
+        children: [
+          Text(
+            style: baseTextStyle
+              .copyWith(color: designVariables.btnLabelAttLowIntInfo),
+            textAlign: TextAlign.end,
+            zulipLocalizations.savingMessageEditLabel),
+          // TODO instead place within outer padding; see Figma:
+          //   https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4026-8775&m=dev
+          LinearProgressIndicator(
```

Comment:
I don't quite follow this comment. Which padding should this be placed within?

Reply:
Here are screenshots of the Figma with and without that outer padding highlighted:

<img width="1058" alt="image" src="https://github.com/user-attachments/assets/c305488a-0d39-44b7-ae4e-495661d4f4ea" />

<img width="1077" alt="image" src="https://github.com/user-attachments/assets/46b67457-547b-4978-8bf3-6b7d18861aaa" />

The progress indicator sits inside that outer padding.

I think it looks fine as-is and would be OK to remove the TODO, revisiting if needed in the future.

Reply:
Hmm, those screenshots don't really help. :slightly_smiling_face: How does this version differ from what's in Figma?

Looking into it a bit more, I think you're referring to the 4px of bottom padding on the MessageWithPossibleSender.

So in the Figma design, the indicator encroaches on that padding and gets closer to whatever's below (like the next message, or a date separator). In this version, when the indicator appears, it instead makes the message taller and stays 4px away from the sibling below.

I see the appeal of the version in Figma, but the difference is subtle. Happy to leave it as a TODO.